pom.xml

```diff
@@ -553,19 +563,34 @@
                     <artifactId>jsonschema2pojo-maven-plugin</artifactId>
                     <version>${jsonschema2pojo.version}</version>
                     <configuration>
+                        <addCompileSourceRoot>true</addCompileSourceRoot>
                         <annotationStyle>none</annotationStyle>
                         <customAnnotator>org.apache.streams.plugins.JuneauPojoAnnotator</customAnnotator>
                         <generateBuilders>true</generateBuilders>
                         <includeGeneratedAnnotation>false</includeGeneratedAnnotation>
-                        <sourceDirectory>${project.basedir}/src/main/jsonschema</sourceDirectory>
-                        <outputDirectory>${project.build.directory}/generated-sources/pojo</outputDirectory>
                     </configuration>
                     <executions>
                         <execution>
-                            <phase>generate-sources</phase>
+                            <id>generate-sources</id>
+                            <phase>none</phase>
+                            <goals>
+                                <goal>generate</goal>
+                            </goals>
+                            <configuration>
+                                <sourceDirectory>${project.basedir}/src/main/jsonschema</sourceDirectory>
+                                <outputDirectory>${project.build.directory}/generated-sources/pojo</outputDirectory>
+                            </configuration>
+                        </execution>
+                        <execution>
+                            <id>generate-test-sources</id>
+                            <phase>none</phase>
                             <goals>
```

Comment:
these phase=none let the parent pom do all the configuration and child modules enable one or both executions by settings phase!=none